tap_s3_csv/s3.py

```diff
@@ -499,3 +500,185 @@ def get_file_handle(config, s3_path):
     s3_bucket = s3_client.Bucket(bucket)
     s3_object = s3_bucket.Object(s3_path)
     return s3_object.get()['Body']
+
+
+class EOLType(Enum):
+    LF = 1
+    CR = 2
+    CRLF = 3
+
+
+class GetFileRangeStream:
+    def __init__(self, bucket: str, key: str, start_byte: int, end_byte: int, chunk_size: int):
+        if (start_byte >= end_byte):
+            raise ValueError(
+                f'start and end byte range is invalid')
+
+        self.bucket = bucket
+        self.key = key
+        self.start_byte = start_byte
+        self.end_byte = end_byte
+        self.chunk_size = chunk_size
+
+    def iter_lines(self):
+        # get file size
+        file_size = self.__get_content_length__()
+        LOGGER.info(f'total file_size: {file_size}')
+
+        if (self.start_byte > file_size):
+            raise ValueError(
+                f'start byte is greater than file size {file_size}')
+
+        # always return header first, after this each chunk iteration will discard first row but include partial row at the end
+        headers = self.__get_headers__()
+        LOGGER.info(headers)
+        yield headers
+
+        eol = self.__get_eol__()
+
+        end = min(self.end_byte, file_size)
```

Comment:
should this be `end = min(self.end_byte, file_size - 1)`? if file_size was 10 bytes, isn't 9 the end byte?